### PR TITLE
Convert infinite values to NaN

### PR DESCRIPTION
**Pull request by yha**
Ref #1261. The discussion there includes some other ideas for how to plot infinite values, so maybe should be left open.
An example which plots incorrectly on GR before this change:
```
plot([1,Inf,2,3])
```
Corresponding reference image: JuliaPlots/PlotReferenceImages.jl#49

### CODE REVIEW

File: src/series.jl [hunk 1]
```diff
@@ -14,7 +14,10 @@ const SeriesData = Union{AVec{<:DataPoint}, Function, Surface, Volume}
 
 prepareSeriesData(x) = error("Cannot convert $(typeof(x)) to series data for plotting")
 prepareSeriesData(::Nothing) = nothing
-prepareSeriesData(s::SeriesData) = handlemissings(s)
+prepareSeriesData(s::SeriesData) = handleinfinites(handlemissings(s))
+
+handleinfinites(s) = s
+handleinfinites(s::AbstractArray{<:MaybeNumber}) = [isinf(x) ? NaN : x for x in s]
```
Comment (mkborregaard) [inline src/series.jl]: This copies the input data. We already do that in one place, in they `computexy` functions. Could we handle that there so we don't copy twice?
Alternatively, given that we do copy the input data, it should be safe to use `replace!` here.

File: src/series.jl [hunk 2]
```diff
@@ -14,7 +14,10 @@ const SeriesData = Union{AVec{<:DataPoint}, Function, Surface, Volume}
...
 
 handlemissings(v) = v
 handlemissings(v::AbstractArray{<:MaybeNumber}) = replace(v, missing => NaN)
```
Comment (mkborregaard) [inline src/series.jl]: and maybe here too?

**Comment by yha** [comment]
We can't rely on `compute_xyz` making a copy, because it's only called after `convertToAnyVector`.
Also, we can't just `replace!`, since `NaN`s may not be compatible with the vector type, and we also want to remove `Missing` from the type.
The new version of this PR copies only once in `prepareSeriesData`, but there's still additional copying in `compute_xyz`. Also, I realized that functions returning infinities or `missing`s can't be handled in this design. I think both these problems can be fixed by handling `plot(::Function, x)` entirely in a recipe rather with `compute_xyz`, but that should probably be another PR.

**Comment by mkborregaard** [comment]
But `ConvertToAnyVector` is always called, no?

**Comment by mkborregaard** [comment]
> since NaNs may not be compatible with the vector type

Which Vector type has `Inf` but not `NaN`?

**Comment by yha** [comment]
> But `ConvertToAnyVector` is always called, no?

Yes, but before `compute_xyz`, with the original input vectors. If it doesn't make a copy, it would replace infinities and missings in the original input.

> Which Vector type has `Inf` but not `NaN`?

Probably none. But you could get  a `v::Vector{Union{Int,Missing}}`, and then `replace!(v,missing=>NaN)` would error.

**Comment by mkborregaard** [comment]
>  If it doesn't make a copy, it would replace infinities and missings in the original input

I don't think that would ever happen?

> But you could get a v::Vector{Union{Int,Missing}}, and then replace!(v,missing=>NaN) would error.

We have `Int`s this late in the pipeline?

**Comment by yha** [comment]
Apparently we do have raw input vectors at this point in the pipeline, including `Int`s.
If I change the method in lines 17:19 from `replace!(..., map(float,a))` to `replace!(..., a)` I get
```
plot([1,missing])   # => InexactError: Int64(NaN)
x = [1.0,missing,Inf]
plot(x)
x      # => [1.0, NaN, NaN]
```

**Comment by yha** [comment]
I also updated the example to show `missing`s as well, and test the case of `Vector{Union{Int,Missing}}`.

Incidentally, is there a reason for this check guarding the image comparison tests?
https://github.com/JuliaPlots/Plots.jl/blob/464c0271f026e53037c64a30c3dc3b40e0dd1ec3/test/runtests.jl#L52
I just commented it out to run them on Windows and it seems to work fine, except one failing test.

**Comment by mkborregaard** [comment]
I guess that was back when ImageMagick was used everywhere and flaky

**Comment by daschw** [comment]
We don't want to run the image comparison tests on Appveyor because we already test them on Travis, it takes very long on Appveyor and some plots failed due to different fonts between Windows/Linux or other similar issues.
But I guess we could also use some Appveyor environment variable for this check instead of `Sys.islinux()` - in fact, we should.

**Comment by yha** [comment]
I think so, yes.

**Pull request merged**